perl/shop/t/Basket.t

```diff
@@ -1,319 +1,364 @@
-    ok( !$hExists->{'Error'}, 'exists: no error' );
-    diag $hExists->{'Error'}->{'Message'}."\n" if $hExists->{'Error'};
+    is( scalar @$ahBaskets, 1, 'testCreateBasket: response format' );
 
-    ok( $hExists->{'Path'} eq $Path, 'exists: basket path' );
-    is( $hExists->{'exists'}, $exists, 'exists?' );
+    my $hCreate = $ahBaskets->[0];
+    ok( !$hCreate->{'Error'}, 'testCreateBasket: check error' );
+    is( $hCreate->{'created'}, 1, 'testCreateBasket: created' );
 }
 
-# Create a Basket and check if the creation was successful
-sub testCreateBasket {
-    my ($Basket) = @_;
-
-    my $ahResults = $BasketService->create( [$Basket] )->result;
-    is( scalar @$ahResults, 1, 'create: result count' );
+sub testDeleteLineItem {
+    my $BasketPath = _setupTestBasketInDB();
 
-    my $hCreate = $ahResults->[0];
-    ok( !$hCreate->{'Error'}, 'create: no error' );
-    diag $hCreate->{'Error'}->{'Message'}."\n" if $hCreate->{'Error'};
+    _deleteProductsIfExists( [$Product_alias_1] );
+    _createProductsInDB(     [$Product_in_1] );
+    my $ProductGUID = _fetchProductGuid($Product_alias_1);
 
-    is( $hCreate->{'created'}, 1, 'created?' );
+    my $ProductLineItem = {
+        'GUID'     => $ProductGUID,
+        'Quantity' => 50,
+    };
+    $BasketService->addProductLineItem( $BasketPath, [$ProductLineItem] )->result;
 
-    return $hCreate->{'Path'};
-}
+    my $hBasketInfo = _getInfo($BasketPath);
 
-sub testGetInfoReference {
-    my ( $basketPath, $basketRef ) = @_;
+    my $LineItemAlias = $hBasketInfo->{LineItemContainer}->{'ProductLineItems'}->[0]->{'Alias'};
 
-    my $ahResults = $BasketService->getInfo( [$basketPath], [], [] )->result;
-    is( scalar @$ahResults, 1, 'getInfo result set' );
+    my $ahResultDelete = $BasketService->deleteLineItem( $BasketPath, [$LineItemAlias] )->result;
+    is( scalar @$ahResultDelete, 1, 'testDeleteLineItem: result count' );
+    my $hResultDelete = $ahResultDelete->[0];
+    ok( !$hResultDelete->{'Error'}, 'testDeleteLineItem: check error' );
+    diag $hResultDelete->{'Error'}->{'Message'} . "\n" if $hResultDelete->{'Error'};
 
-    my $hInfo = $ahResults->[0];
-    ok( !$hInfo->{'Error'}, 'getInfo: no error' );
-    diag "Error: $hInfo->{'Error'}\n" if $hInfo->{'Error'};
+    my $LineItemPath = "$BasketPath/LineItemContainer/$LineItemAlias";
+    is( $hResultDelete->{'Path'},    $LineItemPath, 'testDeleteLineItem: line item path' );
+    is( $hResultDelete->{'deleted'}, 1,             'testDeleteLineItem: deleted' );
+}
 
-    my $hLineItemContainer  = $hInfo->{'LineItemContainer'};
-    my $hLineItemContainer2 = $basketRef->{'LineItemContainer'};
-    ok( $hLineItemContainer->{'TaxArea'}    eq $hLineItemContainer2->{'TaxArea'},       'tax area' );
-    ok( $hLineItemContainer->{'TaxModel'}   eq $hLineItemContainer2->{'TaxModel'},      'tax model' );
-    ok( $hLineItemContainer->{'CurrencyID'} eq $hLineItemContainer2->{'CurrencyID'},    'currencyid' );
+sub testExistsBasket {
```

Comment:
Test case only calls the `exists`-method for existing baskets. Maybe a call for a non existing basket should be added.